keras_contrib/applications/densenet.py

```diff
@@ -444,13 +462,20 @@ def __create_dense_net(nb_classes, img_input, include_top, depth=40, nb_dense_bl
         reduction: reduction factor of transition blocks. Note : reduction value is inverted to compute compression
         dropout_rate: dropout rate
         weight_decay: weight decay
+        dilation_rate: an integer or tuple/list of 2 integers, specifying
+            the dilation rate to use for dilated convolution. Can be a
+            single integer to specify the same value for all spatial
+            dimensions.
+        pooling: Data pooling to reduce resolution, one of "avg", "max", None
+        input_shape: Only used for shape inference in fully convolutional networks.
 
     Returns: keras tensor with nb_layers of conv_block appended
     '''
 
-    concat_axis = 1 if K.image_dim_ordering() == "th" else -1
+    concat_axis = 1 if K.image_data_format() == 'channels_first' else -1
 
-    assert (depth - 4) % 3 == 0, "Depth must be 3 N + 4"
+    if depth is not None:
```

Comment:
Can you give me a case when depth is None? As far as I know, depth is never None. Also, I see you have a test case where you build a DenseNet with depth None. How is the None case handled?